hyperspy/_components/arctan.py

```diff
@@ -39,57 +40,35 @@ class Arctan(Component):
     :math:`x_0`   x0 
     ============ =============
 
+
     Parameters
     -----------
     A : float
+        Amplitude parameter. :math:`\lim_{x\to -\infty}f(x)=-A` and 
+        :math:`\lim_{x\to\infty}f(x)=A`
     k : float
+        Slope (steepness of the step). The larger :math:`k`, the sharper the 
+        step.
     x0 : float
+        Center parameter (position of zero crossing :math:`f(x_0)=0`).
 
     """
 
-    def __init__(self, A=1., k=1., x0=1., minimum_at_zero=False):
-        Component.__init__(self, ['A', 'k', 'x0'])
-        self.A.value = A
-        self.A.grad = self.grad_A
-
-        self.k.value = k
-        self.k.grad = self.grad_k
-
-        self.x0.value = x0
-        self.x0.grad = self.grad_x0
-
-        self.minimum_at_zero = minimum_at_zero
-        self._whitelist['minimum_at_zero'] = ('init', minimum_at_zero)
+    def __init__(self, A=1., k=1., x0=1., module=["numpy", "scipy"], **kwargs):
+        # Not to break scripts once we remove the legacy Arctan
+        if "minimum_at_zero" in kwargs:
+            del kwargs["minimum_at_zero"]
+        super(Arctan, self).__init__(
+            expression="A * arctan(k * (x - x0))",
+            name="Arctan",
+            A=A,
+            k=k,
+            x0=x0,
+            position="x0",
+            module=module,
+            autodoc=False,
+            **kwargs,
+        )
 
         self.isbackground = False
-        self.convolved = False
-        self._position = self.x0
-
-    def function(self, x):
-        A = self.A.value
-        k = self.k.value
-        x0 = self.x0.value
-        if self.minimum_at_zero:
-            return A * (math.pi / 2 + np.arctan(k * (x - x0)))
-        else:
-            return A * np.arctan(k * (x - x0))
-
-    def grad_A(self, x):
-        k = self.k.value
-        x0 = self.x0.value
-        if self.minimum_at_zero:
-            return offset + np.arctan(k * (x - x0))
-        else:
-            return np.arctan(k * (x - x0))
-
-    def grad_k(self, x):
-        A = self.A.value
-        k = self.k.value
-        x0 = self.x0.value
-        return A * (x - x0) / (1 + (k * (x - x0)) ** 2)
-
-    def grad_x0(self, x):
-        A = self.A.value
-        k = self.k.value
-        x0 = self.x0.value
-        return -A * k / (1 + (k * (x - x0)) ** 2)
+        self.convolved = True
```

Comment:
Is that perhaps why the BSS tests were failing - if that then changes the values of the artificial EELS datasets?

Reply:
I would actually suggest to remove it, so it falls back to the default - I don't see why this specific component shouldn't be convolved. @francisco-dlp?

> Is that perhaps why the BSS tests were failing - if that then changes the values of the artificial EELS datasets?

No, this is not the reason because the `function` method is used directly and it is not convolution aware (as it should be). However, `minimum_at_zero` was set to True and this may have been the reason?


For information on this, `model.convolved` is False by default and set to True when the `low_loss` is set
https://github.com/hyperspy/hyperspy/blob/226da1e5b88779b720e840d8d791f900f4111385/hyperspy/models/model1d.py#L317-L331

and in the `__call__` method, `component.convolved` is used only if `model.convolved=True`.

https://github.com/hyperspy/hyperspy/blob/226da1e5b88779b720e840d8d791f900f4111385/hyperspy/models/model1d.py#L405-L424

Reply:
@francisco-dlp: what would you recommend here: use the [default value of `convolved`](https://github.com/hyperspy/hyperspy/blob/fd8e637dc4b544c0b2d0892909007ddde2040c59/hyperspy/component.py#L755) or set it to `False`, as it was before this PR.

Reply:
The ``convolved`` attribute (inherited from EELSLab) is there because in EELS you usually (there different opinions on this) convolve everything but the background components. That's why the default value is ``True``. With this in mind, I think that it is right to use the default value.